src/middlewared/middlewared/restful.py

```diff
@@ -535,6 +535,12 @@ def get_post_args(self, req, resp, **kwargs):
                     'errno': errno,
                 })
             resp = web.Response(status=422)
+        except MatchNotFound as e:
```

Comment:
I dont think MatchNotFound necessarily means the object was not found, we use get=True for other parts of the code, not only for the lookup of the specific object we are looking. e.g. we dont want 404 if for some reason MatchNotFound is raised on user.query (/api/v2.0/user/) so I think we still need a flag for the id lookup and handle this exception within Exception (`isinstance(e, MatchNotFound)`)

Reply:
Good catch, have not thought about it. Probably rest controller should pass new option `get_or_404` and we should raise this new exception if only we got `get_or_404`